core/src/main/java/org/wildfly/channel/ChannelSession.java

```diff
@@ -225,17 +246,37 @@ private Channel.ResolveLatestVersionResult findChannelWithLatestVersion(String g
     }
 
     private Map<Channel, List<ArtifactCoordinate>> splitArtifactsPerChannel(List<ArtifactCoordinate> coordinates) {
+        long start = System.currentTimeMillis();
         Map<Channel, List<ArtifactCoordinate>> channelMap = new HashMap<>();
-        for (ArtifactCoordinate coord : coordinates) {
-            Channel.ResolveLatestVersionResult result = findChannelWithLatestVersion(coord.getGroupId(), coord.getArtifactId(),
-                                                                                     coord.getExtension(), coord.getClassifier(), coord.getVersion());
-            ArtifactCoordinate query = new ArtifactCoordinate(coord.getGroupId(), coord.getArtifactId(), coord.getExtension(), coord.getClassifier(), result.version);
-            Channel channel = result.channel;
-            if (!channelMap.containsKey(channel)) {
-                channelMap.put(channel, new ArrayList<>());
-            }
-            channelMap.get(channel).add(query);
+
+        ForkJoinPool customThreadPool = new ForkJoinPool(versionResolutionParallelism);
```

Comment:
you don't need to use a `ForkJoinPool` as you submit a single task.
Internally, the parallel stream will use its own `ForkJoinPool`.

Reply:
I mean my expectation is that the parallel stream would use my custom FJP internally, I will recheck that.

Reply:
I can confirm that the created ForkJoinPool is used by the parallelStream.